bundles/org.eclipse.swt/Eclipse SWT/cocoa/org/eclipse/swt/widgets/Display.java

```diff
@@ -5988,9 +5988,10 @@ static long dialogProc(long id, long sel, long arg0) {
 
 	switch (Selector.valueOf(sel)) {
 		case sel_changeColor_: {
-			ColorDialog dialog = (ColorDialog)OS.JNIGetObject(jniRef[0]);
-			if (dialog == null) return 0;
-			dialog.changeColor(id, sel, arg0);
+			Object object = OS.JNIGetObject(jniRef[0]);
+			if (object instanceof ColorDialog) {
```

Comment:
Given the BREE is Java 17, does SWT allow one to use pattern matching for this?

Reply:
I went for the simplest fix rather than getting into changing the style in this block of code. I could change to the following if required.

```java
if (object instanceof ColorDialog colorDialog) {
    colorDialog.changeColor(id, sel, arg0);
}
```

Reply:
Sounds reasonable to use Java 17 features here.